src/state.cairo

```diff
@@ -70,77 +71,89 @@ pub struct Transaction {
     pub lock_time: u32,
 }
 
-/// Represents an input of a transaction.
+/// Input of a transaction.
 /// https://learnmeabitcoin.com/technical/transaction/input/
-///
-/// NOTE that `txid` and `vout` fields can be resolved via Utreexo set using the TXO index.
 #[derive(Drop, Copy)]
 pub struct TxIn {
     /// The signature script which satisfies the conditions placed in the txo pubkey script
     /// or coinbase script that contains block height (since 227,836) and miner nonce (optional).
     pub script: @ByteArray,
     /// The sequence number of the input.
     pub sequence: u32,
```

Comment:
How do we process the different `sequence` values? Is it part of the transaction processing, or how do we add `locktime` fields and other limitations triggered by this value?

Reply:
`sequence` enables `locktime` and can also specify relative time intervals based on the block height/time when the target output was included. If absolute/relative timelock is enabled (at least one of the inputs has sequence < 0xffffffff) we need to ensure that the transaction can be included in the current block.

That's a good point, we need to include block time to the `OutPoint` in addition to the `block_height` field.